### DISCUSSION TITLE
MQTT Topic Anordnung für Power Meter

### DISCUSSION THREAD

**Discussion by olomberg** [hoylabs/OpenDTU-OnBattery#166] [Ideas]
Hi Helge.

Schaue gerade mittels MQTT Explorer auf meinen Mosquitto. Dort Sehe ich unter "solar" die ganzen Serials (und darunter die Werte), was auch so ok ist. Sammle die Daten für die halbe Nachbarschaft und Freunde damit ein. Daher tummeln sich dort ca. 20 Inverter.

Spiele aktuell mit dem Dynamic Power Limiter und dem Power Meter rum. Dabei ist mir aufgefallen, dass sich das Power Meter auch direkt unter "solar" reinschreibt. Mit mehreren Anlagen würde er also die Daten im Topic "solar/powermeter" überschreiben.
Würde etwas dagegen sprechen, das Topic powermeter eindeutig zu gestalten?? Evtl. als "solar/InverterSerial/powermeter" oder "solar/powermeter/DeviceID". Bei DeviceID müsste man diese noch in der GUI setzen können.

Bin für jeden Vorschlag offen und kann auch gerne testen.

Danke.
Oliver

**Comment by berni2288** [comment]
Hallo @olomberg,
vielleicht hab ich jetzt nicht den allerneuesten Build, aber bei mir ist das Topic so aufgebaut:
```
hm-800-opendtu/powermeter
```

Den DTU-Namen, also hm-800-opendtu kannst du frei wählen und wir unterstützen aktuell nur einen Powermeter pro DTU.
Du kannst das Basis-Topic unter `/settings/mqtt` ändern.
Also Überschreibungen können denke ich nur passieren wenn 2 Geräte das selbe Basis-Topic verwenden?

LG Berni

**Reply by olomberg** [reply]
Hi Berni.

Schaue mir das gerade nochmals an. Bin aktuell auf "9da5be7". Funktioniert bei mir nicht wirklich schön. Wenn ich das Basis Topic verändere, schreibt er mir das "powermeter" richtig darunter. Leider verschiebt er auch die anderen Topics komplett darunter. Ist ja an sich auch so richtig.
Ist nichts, was ich nicht über Telegraf wieder "richtig" in die InfluxDB schreiben kann, aber irgendwie auch nicht richtig schön. Wäre einfacher, wenn das Topic "powermeter" unter der [serial] hängen würde, bzw, wenn man für das "powermeter" ein eigenes Basis-Topic eingeben könnte.

Gruss

Oliver
